### PR TITLE
WIP: Add allSettled mode for Collector class

### PR DESCRIPTION
**Pull request by basi1iscus**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

### CODE REVIEW

File: lib/collector.js [hunk 1]
```diff
@@ -7,17 +7,35 @@ class Collector {
...
+  isRejectable = true;
   #fulfill = null;
   #reject = null;
   #timer = null;
   #cause = null;
 
-  constructor(keys, { exact = true, timeout = 0 } = {}) {
+  constructor(keys, { exact = true, timeout = 0, isRejectable = true } = {}) {
     this.keys = keys;
+    this.isRejectable = isRejectable;
```
Comment (axbuglak) [inline lib/collector.js:18]: We should check if `isRejectable` is boolean

File: lib/collector.js [hunk 2]
```diff
@@ -88,6 +106,13 @@ class Collector {
   }
 }
 
+function getSettled(err = null, value = null) {
```
Comment (axbuglak) [inline lib/collector.js:109]: It shouldn't be a separate function, it should be inside a class `Collector`.  And what must do this function? Function `set` takes the argument `value` not as an object.

Comment (basi1iscus) [reply lib/collector.js:109]: I made getSettled as static method in class  Collector before,  but @tshemsedinov said said that the function is better.
Function should return an object like Promise.allSettled() depending on whether the result is "fulfilled" or "rejected".

File: lib/collector.js (hunk 1 above)
Comment (basi1iscus) [reply lib/collector.js]: Fixed

**Comment by tshemsedinov** [comment]
I do not remember the purpose of all this stuff, please send me link to video and more explanations in private messages @basi1iscus Thanks!

**Comment by tshemsedinov** [comment]
Also need to rebase on master

**Pull request closed without merging**